### PR TITLE
ucm bugs fixes

### PR DESCRIPTION
**Pull request by ankush-maherwal**

### CODE REVIEW

File: site/controllers/item.php [hunk 1]
```diff
@@ -51,7 +83,7 @@ public function edit()
 		}
 
 		// Redirect to the edit screen.
-		$this->setRedirect(JRoute::_('index.php?option=com_tjucm&view=itemform&layout=edit', false));
+		$this->setRedirect(JRoute::_('index.php?option=com_tjucm&view=itemform&layout=edit' . $this->appendUrl, false));
```
Comment (ankush-maherwal) [inline site/controllers/item.php]: done

File: site/controllers/item.php [hunk 2]
```diff
@@ -160,11 +186,6 @@ public function remove()
 
 				$this->setMessage(JText::_('COM_TJUCM_ITEM_DELETED_SUCCESSFULLY'));
 			}
-
-			// Redirect to the list screen.
-			$menu = JFactory::getApplication()->getMenu();
```
Comment (ankush-maherwal) [inline site/controllers/item.php]: done

File: site/models/item.php [hunk 3]
```diff
@@ -117,32 +119,32 @@ public function &getData($id = null)
 					// Check for published state if filter set.
 					if (((is_numeric($published)) || (is_numeric($archived))) && (($table->state != $published) && ($table->state != $archived)))
 					{
-						return JError::raiseError(404, JText::_('COM_TJUCM_ITEM_DOESNT_EXIST'));
+						return JError::raiseError(404, JText::_('COM_TJUCMitem_DOESNT_EXIST'));
```
Comment (ankush-maherwal) [inline site/models/item.php]: done

File: site/models/item.php [hunk 4]
```diff
@@ -117,32 +119,32 @@ public function &getData($id = null)
...
 					{
-						$this->_item->params->set('access-view', true);
+						$this->item->params->set('access-view', true);
 					}
 				}
 			}
 			else
 			{
-				return JError::raiseError(404, JText::_('COM_TJUCM_ITEM_DOESNT_EXIST'));
+				return JError::raiseError(404, JText::_('COM_TJUCMitem_DOESNT_EXIST'));
```
Comment (ankush-maherwal) [inline site/models/item.php]: done

File: site/models/item.php [hunk 5]
```diff
@@ -284,15 +286,31 @@ public function publish($id, $state)
...
+
+			return $table->delete($id);
+		}
+		else
+		{
+			$app->enqueueMessage(JText::_('COM_TJUCM_ERROR_MESSAGE_NOT_AUTHORISED'), 'error');
+			$app->setHeader('status', 403, true);
+
+			throw new Exception(JText::_('JERROR_ALERTNOAUTHOR'), 403);
+		}
```
Comment (ankush-maherwal) [inline site/models/item.php]: done

File: site/models/itemform.php [hunk 6]
```diff
@@ -553,19 +553,33 @@ public function duplicate(&$pks)
...
 
-		if ($table->delete($data['id']) === true)
+		$ucmTypeId = $this->getState('ucmType.id');
+		$user = JFactory::getUser();
+		$canDelete = $user->authorise('core.type.deleteitem', 'com_tjucm.type.' . $ucmTypeId);
+
+		if ($canDelete)
 		{
-			$this->deleteExtraFieldsData($data['id'], $data['client']);
+			if ($table->delete($data['id']) === true)
```
Comment (ankush-maherwal) [inline site/models/itemform.php]: done

File: site/models/items.php [hunk 7]
```diff
@@ -94,7 +86,41 @@ protected function populateState($ordering = null, $direction = null)
 		$app->setUserState($this->context . '.list', $list);
 		$app->input->set('list', null);
 		$type_id = $app->input->get('id', "", "INT");
-		$this->setState("type_id", $type_id);
+		$this->setState("ucmType.id", $type_id);
```
Comment (ankush-maherwal) [inline site/models/items.php]: done

File: administrator/views/item/tmpl/edit.php [hunk 8]
```diff
@@ -19,34 +19,33 @@
...
+		else
+		{
+			if (task != 'item.cancel' && document.formvalidator.isValid(document.id('item-form')))
+			{
 				Joomla.submitform(task, document.getElementById('item-form'));
 			}
-			else {
+			else
+			{
 				alert('<?php echo $this->escape(JText::_('JGLOBAL_VALIDATION_FORM_FAILED')); ?>');
```
Comment (twsvaishali) [inline administrator/views/item/tmpl/edit.php]: joomla.jText?

**Pull request merged**